src/librustc/mir/cache.rs

```diff
@@ -1,73 +0,0 @@
-use rustc_data_structures::indexed_vec::IndexVec;
-use rustc_data_structures::sync::{RwLock, MappedReadGuard, ReadGuard};
-use rustc_data_structures::stable_hasher::{HashStable, StableHasher, StableHasherResult};
-use rustc_serialize::{Encodable, Encoder, Decodable, Decoder};
-use crate::ich::StableHashingContext;
-use crate::mir::{Body, BasicBlock};
-
-#[derive(Clone, Debug)]
-pub struct Cache {
-    predecessors: RwLock<Option<IndexVec<BasicBlock, Vec<BasicBlock>>>>
-}
-
-
-impl rustc_serialize::Encodable for Cache {
```

Comment:
@oli-obk do I need to worry about the (de)serialization of the cache now that it's part of `Body` and not its own type?

Reply:
`mir::Body` derives the serialization stuff, so it'll pick up the field

Reply:
hmm... oh, the idea is to not serialize it, but regenerate it on deserialization?

Reply:
Yes, that seems to be what the previous implementation did.

Reply:
This is back in place, so this thread isn't needed anymore.